app/src/main/java/com/github/se/eventradar/model/repository/user/FirebaseUserRepository.kt

```diff
@@ -160,6 +180,47 @@ class FirebaseUserRepository(db: FirebaseFirestore = Firebase.firestore) : IUser
     }
   }
 
+  override suspend fun generateQRCode(userId: String): Resource<String> { // upload this in firebase
+    return try {
+      val qrCodeWriter = QRCodeWriter()
+      val bitMatrix = qrCodeWriter.encode(userId, BarcodeFormat.QR_CODE, 200, 200)
+      val bitmap = Bitmap.createBitmap(200, 200, Bitmap.Config.RGB_565)
+      for (x in 0 until 200) {
+        for (y in 0 until 200) {
+          bitmap.setPixel(
+              x,
+              y,
+              if (bitMatrix[x, y]) Color(0xFF000000).hashCode() else Color(0xFFFFFFFF).hashCode())
+        }
+      }
+      // Convert the bitmap to a byte array
+      val baos = ByteArrayOutputStream()
+      bitmap.compress(Bitmap.CompressFormat.PNG, 100, baos)
+      val data = baos.toByteArray()
+
+      // Create a reference to the file in Firebase Storage
+      val storageRef = FirebaseStorage.getInstance().reference
+      val qrCodesRef = storageRef.child("QR_Codes/$userId.png")
+
+      // Upload the file to Firebase Storage
+      val uploadTask = qrCodesRef.putBytes(data)
+      // Get the download URL of the image
+
+      val urlTask =
```

Comment:
why not just add a listener to the above task? (if there's a valid reason please let me know)